app/services/PayPalService.scala

```diff
@@ -0,0 +1,122 @@
+package services
+
+import lib.okhttp.RequestRunners
+import com.netaporter.uri.QueryString
+import com.netaporter.uri.Uri.parseQuery
+import com.typesafe.scalalogging.LazyLogging
+import config.PayPalConfig
+import lib.okhttp.RequestRunners.FutureHttpClient
+import okhttp3.{FormBody, Request, Response}
+import services.paypal.{PayPalBillingDetails, Token}
+import services.touchpoint.TouchpointService
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+
+class PayPalService(apiConfig: PayPalConfig, client: FutureHttpClient) extends TouchpointService with LazyLogging {
+
+  val config = apiConfig
+  // The parameters sent with every NVP request.
+  val defaultNVPParams = Map(
+    "USER" -> config.user,
+    "PWD" -> config.password,
+    "SIGNATURE" -> config.signature,
+    "VERSION" -> config.NVPVersion
+  )
+
+  // Logs the result of the PayPal NVP request.
+  private def logNVPResponse(response: QueryString) = {
+
+    def msg(status: String) = s"PayPal: $status (NVPResponse: $response)"
+
+    retrieveNVPParam(response, "ACK") match {
+      case "Success" => logger.info("Successful PayPal NVP request")
+      case "SuccessWithWarning" => logger.warn(msg("Warning"))
+      case "Failure" => logger.error(msg("Error"))
```

Comment:
Why are you writing the log status level to the log file, does the logger not do that for you?